src/transformers/trainer.py

```diff
@@ -1524,34 +1514,36 @@ def _save_tpu(self, output_dir: Optional[str] = None):
         # They can then be reloaded using `from_pretrained()`
         xm.rendezvous("saving_checkpoint")
         if not isinstance(self.model, PreTrainedModel):
-            if isinstance(_model_unwrap(self.model), PreTrainedModel):
-                if xm.is_master_ordinal():
-                    _model_unwrap(self.model).config.save_pretrained(output_dir)
+            if isinstance(unwrap_model(self.model), PreTrainedModel):
+                unwrap_model(self.model).save_pretrained(
```

Comment:
If `_save_function` is None this can at most save the config but never the model no? Should we maybe just call `unwrap(model).config.save_pretrained(...)` here? Also what is the use case of not saving the model but the config? When do we need to do this?